github/github.go

```diff
@@ -735,9 +735,12 @@ func category(path string) rateLimitCategory {
 // Deprecated: RateLimit is deprecated, use RateLimits instead.
 func (c *Client) RateLimit() (*Rate, *Response, error) {
 	limits, resp, err := c.RateLimits()
-	if limits == nil {
+	if err != nil {
 		return nil, nil, err
 	}
+	if limits == nil {
+		return nil, nil, fmt.Errorf("RateLimits returned nil limits and error. Unable to extract Core rate limit")
```

Comment:
I think we should return original `resp` instead of `nil` here. It's either going to be nil or some valid response. If there's an error, and a valid response, it's better to return that to the caller so they can look into why the error happened, rather than dropping the response information.

Here, and in the if check above.